### PR TITLE
Feat/secret format alignment

### PR DESCRIPTION
**Pull request by yannickstruyf3**
**What this PR does / why we need it**:
Changes the secret format used by CAPX. Uses Prism-go for credential types.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
N/A

**How Has This Been Tested?**:
```
LABEL_FILTERS="capx-feature-test" make test-e2e-calico
```
```
Ran 11 of 22 Specs in 1276.638 seconds
SUCCESS! -- 11 Passed | 0 Failed | 0 Pending | 11 Skipped
PASS
```

**Special notes for your reviewer**:
N/A

**Release note**:

```release-note
- Modified CAPX secret format
```

### CODE REVIEW

**Review by tuxtof — DISMISSED** [review]
LGTM

File: go.mod [hunk 1]
```diff
@@ -4,13 +4,14 @@ go 1.19
 
 require (
 	github.com/blang/semver v3.5.1+incompatible
-	github.com/nutanix-cloud-native/prism-go-client v0.2.0
+	github.com/nutanix-cloud-native/prism-go-client v0.2.1-0.20220926172029-0d345a71515e
```
Comment (thunderboltsid) [inline go.mod:7]: let's get a `v0.3.0` going so you can use that here.

Comment (thunderboltsid) [reply go.mod:7]: this is now ready to use

File: controllers/suite_test.go [hunk 2]
```diff
@@ -35,8 +36,12 @@ import (
 // These tests use Ginkgo (BDD-style Go testing framework). Refer to
 // http://onsi.github.io/ginkgo/ to learn more about Ginkgo.
 
-var k8sClient client.Client
-var testEnv *envtest.Environment
+var (
+	k8sClient client.Client
+	cfg       *rest.Config
+	testEnv   *envtest.Environment
+	err       error
```
Comment (thunderboltsid) [inline controllers/suite_test.go:43]: please move err back in test function scope

File: controllers/suite_test.go [hunk 3]
```diff
@@ -35,8 +36,12 @@ import (
 // These tests use Ginkgo (BDD-style Go testing framework). Refer to
 // http://onsi.github.io/ginkgo/ to learn more about Ginkgo.
 
-var k8sClient client.Client
-var testEnv *envtest.Environment
+var (
+	k8sClient client.Client
+	cfg       *rest.Config
```
Comment (thunderboltsid) [inline controllers/suite_test.go:41]: please move cfg back in test function scope

**Review by tuxtof — DISMISSED** [review]

**Review by thunderboltsid — DISMISSED** [review]

File: config/manager/manager.yaml [hunk 4]
```diff
@@ -6,14 +6,42 @@ metadata:
   name: system
 ---
 apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: nutanix-endpoint
+  namespace: capi-nutanix-system
+data:
+  prismCentral: |-
+    {
```
Comment (deepakm-ntnx) [inline config/manager/manager.yaml:15]: since this is a json, should we be ideally adding a version format telling the schema version?

Comment (deepakm-ntnx) [reply config/manager/manager.yaml:15]: also, we are going away from strong typing to a loose json format.

**Review by deepakm-ntnx — APPROVED** [review]
we will need to modify eksa templates as well

File: config/manager/manager.yaml (hunk 4 above)
Comment (yannickstruyf3) [reply config/manager/manager.yaml:15]: This is only for manager component. Workload cluster specs are unchanged.

**Review by tuxtof — APPROVED** [review]

**Pull request merged**